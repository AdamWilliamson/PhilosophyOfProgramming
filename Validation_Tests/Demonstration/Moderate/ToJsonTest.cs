//using ApprovalTests;
//using System.Collections.Generic;
//using Validations;
//using Validations.Utilities;
//using Xunit;

//namespace Validations_Tests.Demonstration.Moderate
//{
//    public class ToJsonTest
//    {
//        [Fact]
//        public void AlbumValidator_Describe_ToJson()
//        {
//            // Arrange
//            var validator = new AlbumValidator();
//            var runner = new ValidationRunner<Album>(new List<IScopedValidator<Album>>() { validator });

//            // Act
//            var results = runner.Describe();
//            var json = new ErrorMessagePrettifier().ToJson(results);

//            // Assert
//            Approvals.VerifyJson(json);
//        }

//        [Fact]
//        public void BasicValidator_Validate_ToJson()
//        {
//            // Arrange
//            var validator = new AlbumValidator();
//            var runner = new ValidationRunner<Album>(new List<IScopedValidator<Album>>() { validator });

//            // Act
//            var results = runner.Validate(new Album());
//            var json = new ErrorMessagePrettifier().ToJson(results);

//            // Assert
//            Approvals.VerifyJson(json);
//        }
//    }
//}
