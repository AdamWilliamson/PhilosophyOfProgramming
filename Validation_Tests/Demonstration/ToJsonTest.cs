using ApprovalTests;
using System.Collections.Generic;
using Validations;
using Validations.Utilities;
using Xunit;

namespace Validations_Tests.Demonstration
{
    public class ToJsonTest
    {
        [Fact]
        public void BasicValidator_Describe_ToJson()
        {
            // Arrange
            var validator = new BasicValidator();
            var runner = new ValidationRunner<BasicValidatableObject>(new List<IValidator<BasicValidatableObject>>() { validator });

            // Act
            var results = runner.Describe();
            var json = new ErrorMessagePrettifier().ToJson(results);

			// Assert
			Approvals.VerifyJson(json);
        }

		[Fact]
		public void BasicValidator_Validate_ToJson()
		{
			// Arrange
			var validator = new BasicValidator();
			var runner = new ValidationRunner<BasicValidatableObject>(new List<IValidator<BasicValidatableObject>>() { validator });

			// Act
			var results = runner.Validate(new BasicValidatableObject());
			var json = new ErrorMessagePrettifier().ToJson(results);

			// Assert
			Approvals.VerifyJson(json);
		}
	}
}
