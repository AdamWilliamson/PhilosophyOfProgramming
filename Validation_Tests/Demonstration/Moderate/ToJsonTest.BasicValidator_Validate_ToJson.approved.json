{
    "fieldErrors": [
        {
            "property": "TrackNumber",
            "errors": [
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                }
            ]
        },
        {
            "property": "TrackName",
            "errors": []
        },
        {
            "property": "Duration",
            "errors": []
        },
        {
            "property": "IntegerList",
            "errors": []
        },
        {
            "property": "IntegerList[0]",
            "errors": []
        },
        {
            "property": "IntegerList[1]",
            "errors": [
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                },
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                }
            ]
        },
        {
            "property": "IntegerList[2]",
            "errors": [
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                },
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                }
            ]
        },
        {
            "property": "IntegerList[3]",
            "errors": [
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                },
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                }
            ]
        },
        {
            "property": "IntegerList[4]",
            "errors": [
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                },
                {
                    "error": "Is not equal to 1",
                    "isFatal": false,
                    "childErrors": []
                }
            ]
        }
    ]
}